enclave_ops/deployment/dkeyserver/App/socket_server.cpp

```diff
@@ -373,12 +375,22 @@ int sp_ra_proc_msg1_req(const sample_ra_msg1_t *p_msg1,
             break;
         }
 
+        std::string priv = PRIV_RESULT;
+        std::stringstream priv_stream(priv);
+        std::string segment;
+        /* Reversely set 32B EC private key due to little endian sample_ec256_private_t */
+        uint32_t i = SAMPLE_ECP256_KEY_SIZE - 1;
+        while (std::getline(priv_stream, segment, ':') && i < SAMPLE_ECP256_KEY_SIZE) {
+            g_sp_priv_key.r[i] = stoi(segment, 0, 16);
+            i--;
+        }
+
```

Comment:
same here, put it into a separate function?

Reply:
all updated